coverage/docs/design.md

```diff
@@ -0,0 +1,113 @@
+#Overview
+This code coverage tool has two major features.
+1. As a pre-submit tool, it runs code coverage on every single commit to Github and reports coverage change back to the PR as a comment by a robot account. It also has the ability to block a PR from merging if coverage falls below threshold
+2. As a post-submit / periodical running job, it reports on TestGrid to show users how coverage changes over time.
+
+##Users
+The presubmit tool is intended for a developer to see the impact on code coverage of his/her commit. It can also be used by repo managers to block any PR from merging if the coverage falls under customized threshold.
+
+The periodical testgrid report is for repo managers and/or test infra team to monitor code coverage stats over time.
+
+##Limitation
+As of now, the code coverage tool only collect code coverage for Go files. The support for more programming languages may be added later
+
+##Background - Prow
```

Comment:
Not sure this is necessary. You look to be developing a standalone tool that may or may not be run inside of a Prow job.

Reply:
Done. Deleted